Solaris 11.3 does not have arc4random(3), so we use getentropy(2)

--- rustc-1.84.0-src/library/std/src/sys/random/mod.rs.~1~	2025-01-09 16:46:52.279256314 +0000
+++ rustc-1.84.0-src/library/std/src/sys/random/mod.rs	2025-01-09 16:57:47.227182714 +0000
@@ -18,12 +18,14 @@
         target_os = "netbsd",
         target_os = "openbsd",
         target_os = "rtems",
-        target_os = "solaris",
         target_os = "vita",
     ))] {
         mod arc4random;
         pub use arc4random::fill_bytes;
-    } else if #[cfg(target_os = "emscripten")] {
+    } else if #[cfg(any(
+        target_os = "emscripten",
+        target_os = "solaris",
+    ))] {
         mod getentropy;
         pub use getentropy::fill_bytes;
     } else if #[cfg(target_os = "espidf")] {
